### PR TITLE
bpo-35488: Add test for ** glob matching in pathlib

### PR DESCRIPTION
**Pull request by tonybaloney**
See https://bugs.python.org/issue35488

pathlib Path.match does not support glob-style ** recursive pattern matching.

Add a test to verify this behaviour as unsupported. This will also follow a documentation PR.

See also https://bugs.python.org/issue29249

https://bugs.python.org/issue35488

### CODE REVIEW

File: Lib/test/test_pathlib.py
Comment (tirkarthi) [inline Lib/test/test_pathlib.py:349]: Thanks for the PR. Is it worth adding `self.assertTrue(P('/a/b/c.py').match('/a/**/*.py'))` where 
* `c.py` matches with `*.py`
* `b` matches with `**`
* `a` matches with `a`

Internally the function first checks for the length of the split parts to be equal along with transforming '**' as below that matches with 'b' : 

```python
>>> import fnmatch
>>> fnmatch.translate('**')
'(?s:.*.*)\\Z'
```

I don't know if this can be documented or we can just leave it without mentioning the support of this since it requires documenting the internal implementation to say how this match occurs. It's just a suggestion though. 

I will wait for review from @serhiy-storchaka with respect to https://bugs.python.org/msg285311

File: Lib/test/test_pathlib.py
Comment (brettcannon) [inline Lib/test/test_pathlib.py]:
```suggestion
        # Multi-part glob-style pattern.
```

File: Lib/test/test_pathlib.py
Comment (brettcannon) [reply Lib/test/test_pathlib.py:349]: More tests don't hurt when they aren't overly restrictive and this idea doesn't strike me as such.

File: Lib/test/test_pathlib.py
Comment (tonybaloney) [reply Lib/test/test_pathlib.py]: None of the other single-line comments in this file have periods? I can update all of them or just this one?

File: Lib/test/test_pathlib.py
Comment (tonybaloney) [reply Lib/test/test_pathlib.py:349]: committed in https://github.com/python/cpython/pull/11171/commits/9688d5385fa31bb2aa7dc59a730c73e3f45584e3

File: Lib/test/test_pathlib.py
Comment (brettcannon) [reply Lib/test/test_pathlib.py]: You can update all of them since this isn't code and it's easy to see what changed.

Comment (tonybaloney) [reply Lib/test/test_pathlib.py]: @brettcannon 👍 done in https://github.com/python/cpython/pull/11171/commits/0e2800a4e5c564640650722ccb4641f34dd6b090

**Review by brettcannon — APPROVED** [review]

**Comment by brettcannon** [comment]
@tonybaloney can you add a news entry for your changes? You can use https://blurb-it.herokuapp.com/ to do it entirely in the browser. Otherwise LGTM!

**Comment by tonybaloney** [comment]
@brettcannon done!

**Comment by brettcannon** [comment]
Thanks!

**Pull request merged**
